### PR TITLE
Feature/queue submit

### PR DESCRIPTION
**Pull request by LeonBrands**
Added Queue::submit() and Queue::waitIdle() with a sample and unit tests

### CODE REVIEW

File: legion/engine/llri-dx/detail/instance.cpp [hunk 1]
```diff
@@ -271,22 +271,35 @@ namespace LLRI_NAMESPACE
             const D3D12_COMMAND_LIST_TYPE type = internal::mapQueueType(queueDesc.type);
 
             std::vector<void*> queues(desc.adapter->m_nodeCount, nullptr);
+            std::vector<Fence*> fences(desc.adapter->m_nodeCount, nullptr);
```
Comment (GlynLeine) [inline legion/engine/llri-dx/detail/instance.cpp:274]: see discord

File: legion/engine/llri-dx/detail/queue.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace LLRI_NAMESPACE
+{
+    result Queue::impl_submit(const submit_desc& desc)
+    {
+        HRESULT r;
+        unsigned long index;
+
+        if (desc.nodeMask != 0)
+            _BitScanForward64(&index, desc.nodeMask);
```
Comment (GlynLeine) [inline legion/engine/llri-dx/detail/queue.cpp:18]: this is msvc specific, but i don't know if it still works with clang on windows? Maybe try compiling it with clang on windows. if it compiles then it's all fine.

Comment (LeonBrands) [reply legion/engine/llri-dx/detail/queue.cpp:18]: I believe I tested this on ClangCL but I can check later today

Comment (LeonBrands) [reply legion/engine/llri-dx/detail/queue.cpp:18]: Tested on both MSVC and ClangCL and all unit tests pass

**Review by GlynLeine — APPROVED** [review]

**Pull request merged**